app/templates/main/map.html

```diff
@@ -0,0 +1,72 @@
+<html>
+<head>
+<script src="http://maps.googleapis.com/maps/api/js"></script>
+<script>
+geocoder = new google.maps.Geocoder();
+address = document.getElementById("address");
+function initialize() {
+var initial_coords = new google.maps.LatLng(39.952, -75.195);
+var mapProp = {
+center: initial_coords,
+zoom:17,
+mapTypeId: google.maps.MapTypeId.ROADMAP
+};
+map=new google.maps.Map(document.getElementById("googleMap"),mapProp);
+}
+function reinitialize() {
+var initial_coords = new google.maps.LatLng(39.952, -75.195);
+var mapProp = {
+center: initial_coords,
+zoom:17,
+mapTypeId: google.maps.MapTypeId.ROADMAP
+};
+var map=new google.maps.Map(document.getElementById("googleMap"),mapProp);
+if (geocoder) {
+console.log(address);
+geocoder.geocode( { 'address': address}, function(results, status) {
+if (status == google.maps.GeocoderStatus.OK) {
+if (status != google.maps.GeocoderStatus.ZERO_RESULTS) {
+map.setCenter(results[0].geometry.location);
+}
+else {
+alert("Address not found!");
+}
+}
+else {
+alert("Geocode not successful - " + status);
```

Comment:
Instead of nesting these if-statements, could you do them sequentially? Or at least modify the logic s.t. the flash/alert logic directly follows the conditional.
e.g.
```
if !(geocoderStatus) {
    alert("Geocode not successful")
}

if!(zeroResults) {
    alert("No address")
}
```

if they must be nested, then put the error alert _before_ entering a new conditional